crates/cxx-qt-gen/src/generator/naming/property.rs

```diff
@@ -49,24 +54,41 @@ pub struct QPropertyNames {
     pub reset: Option<Name>,
 }
 
-impl From<&ParsedQProperty> for QPropertyNames {
-    fn from(property: &ParsedQProperty) -> Self {
+impl QPropertyNames {
+    pub(crate) fn try_from_property(
+        property: &ParsedQProperty,
+        structured_qobject: &StructuredQObject,
+    ) -> Result<Self> {
         let property_name = property_name_from_rust_name(property.ident.clone());
 
         // Cache flags as they are accessed multiple times
         let flags = &property.flags;
 
-        let getter = NameState::from_flag_with_auto_fn(&flags.read, || {
-            getter_name_from_property(&property_name)
-        });
-
-        let setter = flags.write.as_ref().map(|setter| {
-            NameState::from_flag_with_auto_fn(setter, || setter_name_from_property(&property_name))
-        });
+        let getter = NameState::from_flag_with_auto_fn(
+            &flags.read,
+            || getter_name_from_property(&property_name),
+            structured_qobject,
+        )?;
+
+        let setter = if let Some(setter) = &flags.write {
```

Comment:
Feels like it should be possible to simplify this but can't think of how right now :sweat_smile:

Reply:
My old way of writing those was to use if let and if it could be done with map, clippy usually tells you, I think its not possible here maybe because of the Result? I'm sure there is a way to improve it though

Reply:
You should be able to use `map` with [`transpose`](https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose).
So something like:
```rust
let setter = flags.write.map(|setter| NameState::from_flag_with_auto_fn(
                setter,
                || setter_name_from_property(&property_name),
                structured_qobject,
            )).transpose()?;
```

Reply:
Which allows you to go from `Result<Option<>>` to `Option<Result<>>` and vice-versa

Reply:
Aha nice :-)

Reply:
used it, did need a clone of the flag though I think